### ISSUE TITLE
Bug with attaching BelongsToMany relationships

### ISSUE THREAD

**Issue by mrmonat** [laravel/nova-issues#148]
Nova is trying using the wrong Model relationship while searching for attachable Resources.

Example to Reproduce:

ProductGroup Nova Resource fields:
```
public function fields(Request $request)
{
        return [
               ...
               BelongsToMany::make('ProductFamilies', 'families', ProductFamily::class)->searchable(),
        ];
}

```

ProductFamily Eloquent Model Relationship:
```
public function families()
{
       return $this->belongsToMany(ProductFamily::class);
}
```

Now the Problem is visible in ```Laravel\Nova\Http\Controllers\AttachableController```'s index method:
```
public function index(NovaRequest $request)
{
        $field = $request->newResource()
                    ->availableFields($request)
                    ->firstWhere('attribute', $request->field); 
     
        ...

}
```
```$request->field``` is 'product-families' but should be 'families'

**Comment by alexjose** [comment]
This issue seemed to be fixed in the 1.0.6 release.

**Comment by mrmonat** [comment]
No, its not fixed. I just tried it with a fresh installation of Laravel with fresh copy of nova 1.0.6

I also noticed that the bug is only Present, if the Resource you want to attach to your Resource has a two or more word long name.

The Problem is in ```Laravel\Nova\Http\Controllers\AttachableController```'s index method. I just did some debugging and noticed that ```->firstWhere('attribute', $request->field);``` only works for "one word" Resources.

Example of what I noticed in the ```index``` method:

My resource has a "two word" name: ProductFamily

```$request->field```: 'product-families'
```->availableFields($request)```: ```[{"component":"text-field","prefixComponent":true,"indexName":"ID","name":"ID","attribute":"id","value":null,"panel":null,"sortable":true,"textAlign":"left"},{"component":"text-field","prefixComponent":true,"indexName":"Name","name":"Name","attribute":"name","value":null,"panel":null,"sortable":true,"textAlign":"left"},{"component":"belongs-to-many-field","prefixComponent":true,"indexName":"Product Family","name":"Product Family","attribute":"families","value":null,"panel":null,"sortable":false,"textAlign":"left","resourceName":"product-families","belongsToManyRelationship":"families","searchable":false,"listable":true}]```

**Comment by zhouchang2017** [comment]
+1
```
exception:"ErrorException"
file:"/Users/z/Documents/code/bestkit-erp-server-nova/vendor/laravel/nova/src/Http/Controllers/AttachableController.php"
line:23
message:"Trying to get property 'resourceClass' of non-object"
```

Models/ProductType
```
public function attributeGroups() {
   return $this->belongsToMany(AttributeGroup::class, 'product_type_attribute_group', 'type_id', 'group_id');
}
```
Models/AttributeGroup
```
public function types() {
   return $this->belongsToMany(ProductType::class,'product_type_attribute_group','group_id','type_id');
}
```

Nova/ProductType
```
public function fields(Request $request) {
        return [
            ...
            BelongsToMany::make('AttributeGroups'),
        ];
}
```

Nova/AttributeGroup
```
public function fields(Request $request) {
        return [
            ...

            BelongsToMany::make('ProductTypes','types'),
        ];
}
```

**Comment by joselara** [comment]
Resource with TwoWords seems to be broken with BelongsToMany.

**Comment by rovansteen** [comment]
This has nothing to do with two words. For example this breaks too:

Model
```php
class Story extends Model
{
    /**
     * The authors that belong to the story.
     */
    public function authors()
    {
        return $this->belongsToMany(User::class);
    }
}
```

Resource
```php
class Story extends Resource
{
    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function fields(Request $request)
    {
        return [
            //
            BelongsToMany::make('Authors', 'authors', User::class),
        ];
    }
}

Gives a 500 on the attach screen.
```
exception: "ErrorException"
file: "/vendor/laravel/nova/src/Http/Controllers/AttachableController.php"
line: 23
message: "Trying to get property of non-object"
```

This part of the `AttachableController` returns null:
```php
        $field = $request->newResource()
                    ->availableFields($request)
                    ->firstWhere('attribute', $request->field);
```

That's because `$request->field` returns `users` rather than `authors`.

**Comment by loren138** [comment]
In `AttachableController`, I replaced lines 18-20 with:
```
        $field = $request->newResource()
                    ->availableFields($request)
                    ->firstWhere('resourceName', $request->field);
```
which enabled viewing the list of resources, but trying to add a resource still failed...

**Comment by rovansteen** [comment]
Yes seems like $request->field is coming from the route so I’m guessing the generated url is wrong?

**Comment by joselara** [comment]
If the resource is a singular name, it's fine.

**Comment by rovansteen** [comment]
Well that doesn’t make sense for a belongs to many relationship? And if Laravel is looking for a relationship users rather than authors (in my case) I fail to see how that is a singular/plural issue.

**Comment by megan-starr9** [comment]
yep, seeing the same!

It looks like the correct value is being passed in the url in the viaRelationship parameter, but that value isn't what is being used as the field to actually load the result resources.

If I put that value directly into AttachableController::index() in place of "$request->field" I get results as expected.

**Comment by meyer59** [comment]
Seems to be the same issue as #76 ?

**Comment by mrmonat** [comment]
@meyer59 yes, sorry too many open issues, didn't see

**Comment by arnordhaenens** [comment]
@mrmonat As described in #76 I was able to resolve a similar issue by overwriting the default implementation of `public static function uriKey()` .

**Comment by mrmonat** [comment]
Fixed in Nova v1.0.8
